### PR TITLE
Initial version of the front end

### PR DESCRIPTION
**Pull request by toon-leader-bacon**
To test the front end, please pull the code, navigate to the `src/main/java/group1/mips_simulator/SimApplication.java` file and run the `public class SimApplication`. Wait a few moments, then you'll see the front end. 

image

Currently, only the input fields, and the ability for force the value into certain registers is implemented. I'll do the load & store once the memory is merged. 
 
~~And I'll work on the `.bi` file part next.~~

Added the `.bi` loading part. Moved the actual implementation to the ROM loader to put it into memory

### CODE REVIEW

File: src/main/java/group1/mips_simulator/SimController.java [hunk 1]
```diff
@@ -1,14 +1,176 @@
...
+
+
+    // endregion
+
+    //region Right Column (Other Registers)
+    public VBox VBoxRight_OtherRegs;
+    public TextField PC;
+    public TextField MAR;
+    public TextField MBR;
+    public TextField RI;
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/SimController.java:72]: im guessing RI is the equivalent of IR (Instruction Register) in spec right? 
may consider a refactor for all the "RI"'s to "IR"

File: src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java [hunk 2]
```diff
@@ -1,66 +1,103 @@
...
+    // and set to 0 if not?)
     Register IR = new Register();       //Instruction Reg : holds instruction to be exe
     Register MAR = new Register();      //Mem Addr Reg : holds address of word to be fetched from memory
     Register MBR = new Register();      //Mem Buffer Reg : holds word just-fetched-from/to-be-writen-to memory
     Register MFR = new Register();      //Mem Fault Reg : contains ID code if machine fault after it occurs
     //Register[] FPR = new Register[2]; //float point register placeholder <<DO NOT IMPLEMENT TILL PART 4>>
     //Other registers add here as we need them (prof said he hasnt given us all of them)
 
-    public RegisterFile(){
+    public RegisterFile() {
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java:31]: its beautiful, nice

Comment (akm303) [reply src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java:31]: not a fan of nullValue tho? haha

**Review by akm303 — APPROVED** [review]
Approved

Awesome stuff dude, i tested it out myself and it looks great
running into some bugs but im sure you're already aware of them or in the process of working them out, just in case I'll detail them in a second message

**Comment by akm303** [comment]
frontend 2024-02-25 at 10 19 49 AM

Minor bug, want to bring attention to it just in case:
I think PC and MAR (and probably other <16b registers), since they are a smaller register size, are truncating the right-most bits when displaying. We'll want them to truncate the left-most bits instead

File: src/main/java/group1/mips_simulator/SimController.java [hunk 3]
```diff
@@ -1,14 +1,176 @@
...
+    public Button GPR3_button;
+
+
+    //endregion
+
+    //region Middle Column (Index Registers)
+    public VBox VBoxCenter_IXRs;
+    public TextField IXR1;
+    public TextField IXR2;
+
```
[the remark this answers is not in the document]
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/SimController.java:58]: Will remove. All nitpicks are appreciated :)

File: src/main/java/group1/mips_simulator/SimController.java (hunk 1 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/SimController.java:72]: Yea it should be IR. This is a typo. Will refactor, good catch

File: src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java (hunk 2 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java:31]: I've got an auto linter installed so every time I save it'll mess with the whitespace. Also, I was getting null exceptions so I decided to fix this so my tests actually worked

**Comment by toon-leader-bacon** [comment]
> frontend 2024-02-25 at 10 19 49 AM
> Minor bug, want to bring attention to it just in case: I think PC and MAR (and probably other <16b registers), since they are a smaller register size, are truncating the right-most bits when displaying. We'll want them to truncate the left-most bits instead

Yea, I need to figure out a way to fix that. But because it's a display issue and not a functional issue it's not a high priority for me.

**Pull request merged**